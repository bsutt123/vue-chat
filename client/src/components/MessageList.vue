<template>
  <div :class="$style.container" >
    <h3 :class="$style.heading" > {{ activeRoom.name }} </h3>
    <ul :class="$style.messageList" >
      <li
        v-for='(message, index) in messages'
        :key="index"
        :class="$style.messageItem"
      >
        {{ message.content }}
      </li>
    </ul>
    <message-creator></message-creator>
  </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex';
import MessageCreator from './MessageCreator';

const { mapState } = createNamespacedHelpers('messages');

export default {
  computed: {
    ...mapState({
      messages: state => state.messages,
    }),
    activeRoom() {
      return this.$store.getters['rooms/activeRoom'];
    },
  },
  components: {
    'message-creator': MessageCreator,
  },
};
</script>

<style lang="postcss" module>


.container {
  position: relative;
}

.heading {
  margin: 1rem 0;
  text-align: center;
  font-size: 2rem;
}

.messageList {
  list-style: none;
  font-size: .8rem;
  margin-bottom: 6rem;

  & > *:not(:last-child) {
    margin-bottom: 1rem;
  }

  & > *:nth-child(odd) {
    background-color: gray(225);
  }
}

.messageItem {
  padding: .25rem 1rem;
}

</style>
