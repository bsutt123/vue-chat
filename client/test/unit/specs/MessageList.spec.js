import { shallow, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VModal from 'vue-js-modal';

import { CssModuleTestHelperMixin } from '../utils/test-helper';
import storeObject from '../utils/testStore';
import MessageList from '../../../src/components/MessageList';


const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VModal);
localVue.use(CssModuleTestHelperMixin);

describe('MessageList', () => {
  let store;
  let wrapper;

  beforeEach( () => {
    store = new Vuex.Store(storeObject);
    wrapper = shallow(MessageList, { localVue, store })
  });

  it('renders the message list', () => {
    expect(wrapper.html()).toMatchSnapshot();
  })
});