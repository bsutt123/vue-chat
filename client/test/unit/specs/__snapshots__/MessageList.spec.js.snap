// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList renders the message list 1`] = `
<div class="container">
    <h3 class="heading"> Room 1 </h3>
    <ul class="messageList">
        <li class="messageItem">

        </li>
        <li class="messageItem">

        </li>
        <li class="messageItem">

        </li>
    </ul>
    <!---->
</div>
`;
